test/extensions/quic_listeners/quiche/active_quic_listener_test.cc

```diff
@@ -221,30 +249,46 @@ TEST_P(ActiveQuicListenerTest, FailSocketOptionUponCreation) {
       .WillOnce(Return(false));
   auto options = std::make_shared<std::vector<Network::Socket::OptionConstSharedPtr>>();
   options->emplace_back(std::move(option));
-  EXPECT_THROW_WITH_REGEX(std::make_unique<ActiveQuicListener>(*dispatcher_, connection_handler_,
-                                                               listen_socket_, listener_config_,
-                                                               quic_config_, options),
-                          EnvoyException, "Failed to apply socket options.");
+  EXPECT_THROW_WITH_REGEX(
+      std::make_unique<ActiveQuicListener>(*dispatcher_, connection_handler_, listen_socket_,
+                                           listener_config_, quic_config_, options, enabledFlag()),
+      EnvoyException, "Failed to apply socket options.");
 }
 
 TEST_P(ActiveQuicListenerTest, ReceiveFullQuicCHLO) {
+  quic::QuicBufferedPacketStore* const buffered_packets =
+      quic::test::QuicDispatcherPeer::GetBufferedPackets(quic_dispatcher_.get());
+  configureQuicRuntimeFlag(/* runtime_enabled = */ true);
   configureMocks(/* connection_count = */ 1);
-  SendFullCHLO(quic::test::TestConnectionId(1));
+  sendFullCHLO(quic::test::TestConnectionId(1));
   dispatcher_->run(Event::Dispatcher::RunType::NonBlock);
+  EXPECT_FALSE(buffered_packets->HasChlosBuffered());
+  EXPECT_FALSE(quic_dispatcher_->session_map().empty());
   ReadFromClientSockets();
 }
 
-TEST_P(ActiveQuicListenerTest, ProcessBufferedChlos) {
-  EnvoyQuicDispatcher* const envoy_quic_dispatcher =
-      ActiveQuicListenerPeer::quic_dispatcher(*quic_listener_);
+/* Not configuring runtime key for `quic.enabled`
+and checking that quic processing is enabled by default. */
```

Comment:
I think we are still configuring quic.enabled as long as enabledFlag() is still called in SetUp(). This test is testing the default behavior of Runtime::MockLoader.